src/main/java/org/broadinstitute/consent/http/models/DACUser.java

```diff
@@ -53,6 +60,41 @@ public DACUser(Integer dacUserId, String email, String displayName,
         this.additionalEmail = additionalEmail;
     }
 
+    /**
+     * Convenience method for backwards compatibility support for older clients.
+     *
+     * @param json A json string that may or may not be correctly structured as a DACUser
+     */
+    public DACUser(String json) {
+        GsonBuilder builder = new GsonBuilder();
+        // Register an adapter to manage the date types as long values
+        builder.registerTypeAdapter(Date.class, (JsonDeserializer<Date>)
+                (json1, typeOfT, context) -> new Date(json1.getAsJsonPrimitive().getAsLong()));
+        Gson gson = builder.create();
+        DACUser u = gson.fromJson(json, DACUser.class);
+        if (u.getDacUserId() != null) {
+            this.setDacUserId(u.getDacUserId());
+        }
+        if (u.getEmail() != null) {
+            this.setEmail(u.getEmail());
+        }
+        if (u.getDisplayName() != null) {
+            this.setDisplayName(u.getDisplayName());
+        }
+        if (u.getCreateDate() != null) {
+            this.setCreateDate(u.getCreateDate());
+        }
+        if (u.getAdditionalEmail() != null) {
+            this.setAdditionalEmail(u.getAdditionalEmail());
+        }
+        if (u.getEmailPreference() != null) {
```

Comment:
Can be empty email, displayName or additionalEmail ? If this is possible, we can validate using !StringUtils.isEmpty

Reply:
Nice catch - will look at improving these checks.